rosbag2_cpp/src/rosbag2_cpp/reindexers/sequential_reindexer.cpp

```diff
@@ -0,0 +1,265 @@
+{
+namespace details
+{
+std::vector<std::string> resolve_relative_paths(
+  const std::string & base_folder, std::vector<std::string> relative_files, const int version = 4)
+{
+  auto base_path = rcpputils::fs::path(base_folder);
+  if (version < 4) {
+    // In older rosbags (version <=3) relative files are prefixed with the rosbag folder name
+    base_path = rcpputils::fs::path(base_folder).parent_path();
+  }
+
+  rcpputils::require_true(
+    base_path.exists(), "base folder does not exist: " + base_folder);
+  rcpputils::require_true(
+    base_path.is_directory(), "base folder has to be a directory: " + base_folder);
+
+  for (auto & file : relative_files) {
+    auto path = rcpputils::fs::path(file);
+    if (path.is_absolute()) {
+      continue;
+    }
+    file = (base_path / path).string();
+  }
+
+  return relative_files;
+}
+}  // namespace details
+
+std::string strip_parent_path(const std::string & relative_path)
+{
+  return rcpputils::fs::path(relative_path).filename().string();
+}
+
+SequentialReindexer::SequentialReindexer(
+  std::unique_ptr<rosbag2_storage::StorageFactoryInterface> storage_factory,
+  std::shared_ptr<SerializationFormatConverterFactoryInterface> converter_factory,
+  std::unique_ptr<rosbag2_storage::MetadataIo> metadata_io)
+: storage_factory_(std::move(storage_factory)),
+  converter_(nullptr),
+  metadata_io_(std::move(metadata_io)),
+  converter_factory_(std::move(converter_factory))
+{}
+
+SequentialReindexer::~SequentialReindexer()
+{
+  reset();
+}
+
+void SequentialReindexer::reset()
+{
+  if (storage_) {
+    storage_.reset();
+  }
+}
+
+
+bool SequentialReindexer::comp_rel_file(
+  const std::string & first_path, const std::string & second_path)
+{
+  std::regex regex_rule(".*_(\\d+)\\.db3", std::regex_constants::ECMAScript);
+
+  std::smatch first_match;
+  std::smatch second_match;
+
+  auto first_regex_good = std::regex_match(first_path, first_match, regex_rule);
+  auto second_regex_good = std::regex_match(second_path, second_match, regex_rule);
+
+  // Make sure the paths have regex matches
+  if (!first_regex_good || !second_regex_good) {
+    throw std::runtime_error("Malformed relative file name. Expected numerical identifier.");
+  }
+
+  // Convert database numbers to uint
+  u_int32_t first_db_num = std::stoul(first_match.str(1), nullptr, 10);
+  u_int32_t second_db_num = std::stoul(second_match.str(1), nullptr, 10);
+
+  return first_db_num < second_db_num;
+}
+
+std::vector<std::string> SequentialReindexer::get_database_files(const std::string & base_folder)
+{
+  // Look in the uri directory to see what database files are there
+  std::vector<std::string> output;
+  for (auto & p_ : boost::filesystem::directory_iterator(base_folder)) {
+    // We are ONLY interested in database files
+    if (p_.path().extension() != ".db3") {
+      continue;
+    }
+
+    output.emplace_back(p_.path().c_str());
+    std::cout << "Found path: " << p_.path().c_str() << "\n";
+  }
```

Comment:
is there any way we can get rid of boost here? I think the iterator is the only place where you currently rely on boost.
Maybe we could come up with a solution with rcpputils.

Reply:
Yeah, that's the only place I use it. I hate having to use a sledgehammer on a thumbtack, lol.

Scanning through `rcpputils::filesystem_helper` to see if there's a more... _Elegant_ solution.